optuna/study/study.py

```diff
@@ -45,6 +45,61 @@
 _logger = logging.get_logger(__name__)
 
 
+def _check_and_convert_to_values(
+    n_objectives: int, original_value: Union[float, Sequence[float]], trial_number: int
+) -> Tuple[Optional[List[float]], Optional[str]]:
+    if isinstance(original_value, Sequence):
+        if n_objectives != len(original_value):
+            return (
+                None,
+                (
+                    f"The number of the values "
+                    f"{len(original_value)} did not match the number of the objectives "
+                    f"{n_objectives}."
+                ),
+            )
+        else:
+            _original_values = list(original_value)
+    else:
+        _original_values = [original_value]
+
+    _checked_values = []
+    for v in _original_values:
+        checked_v, failure_message = _check_single_value(v, trial_number)
+        if failure_message is not None:
+            # TODO(Imamura): Construct error message taking into account all values and do not
+            #  early return
+            # `value` is assumed to be ignored on failure so we can set it to any value.
+            return None, failure_message
+        elif isinstance(checked_v, float):
+            _checked_values.append(checked_v)
+        else:
+            assert False
+
+    return _checked_values, None
+
+
+def _check_single_value(
+    original_value: float, trial_number: int
+) -> Tuple[Optional[float], Optional[str]]:
+    value = None
+    failure_message = None
+
+    try:
+        value = float(original_value)
+    except (
+        ValueError,
+        TypeError,
+    ):
+        failure_message = f"The value {repr(original_value)} could not be  cast to float."
```

Comment:
Perhaps not introduced by this PR? 🤔 
```suggestion
        failure_message = f"The value {repr(original_value)} could not be cast to float."
```

Reply:
✂️